extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1047,10 +1047,26 @@ private static void producePersistenceUnitDescriptorFromConfig(
         descriptor.getProperties().setProperty(AvailableSettings.HBM2DDL_CHARSET_NAME,
                 persistenceUnitConfig.database.charset.name());
 
-        if (persistenceUnitConfig.database.globallyQuotedIdentifiers) {
+        // Quoting options
+        boolean globalQuoting = persistenceUnitConfig.quote.allIdentifiers
+                || persistenceUnitConfig.database.globallyQuotedIdentifiers;
+        if (globalQuoting) {
             descriptor.getProperties().setProperty(AvailableSettings.GLOBALLY_QUOTED_IDENTIFIERS, "true");
         }
 
+        if (globalQuoting && persistenceUnitConfig.quote.skipColumnDefinitions) {
+            descriptor.getProperties().setProperty(
+                    AvailableSettings.GLOBALLY_QUOTED_IDENTIFIERS_SKIP_COLUMN_DEFINITIONS, "true");
+        } else if (persistenceUnitConfig.quote.skipColumnDefinitions) {
+            LOG.warn("Enabling the quoting option 'skip-column-definitions' is useful when 'all-identifiers' is enabled too.");
```

Comment:
If all those properties can only be combined in certain ways, perhaps it would be better to have one property with values that are a bit more explicit than just `true`/`false`?

For example:

```
quarkus.hibernate-orm.quote-identifiers = all
quarkus.hibernate-orm.quote-identifiers = all-skip-column-definitions
quarkus.hibernate-orm.quote-identifiers = keywords
quarkus.hibernate-orm.quote-identifiers = none
```

Note you can define an enum for all the possible values, so that they will automatically be listed in the documentation. See `io.quarkus.hibernate.orm.deployment.HibernateOrmConfigPersistenceUnit.HibernateOrmConfigPersistenceUnitQuery.NullOrdering` for example. You will still need to document the meaning of each value in the javadoc of the configuration property.

---

If you still want to go with this structure for some reason, and feel like warnings are necessary, I'd provide a more straightforward and with the full property keys:

```suggestion
            LOG.warnf("Enabling the quoting option '%s' is not useful because '%s' is disabled.",
                                    HibernateOrmRuntimeConfig.puPropertyKey(persistenceUnitName, "quote.skip-column-definitions"),
                                    HibernateOrmRuntimeConfig.puPropertyKey(persistenceUnitName, "quote.all-identifiers"));
```

Reply:
I had thought about that possibility, but in 6.0 the option to auto-quote identifiers with initial underscore was introduced. Now I've noticed that the option is enabled only by dialects, but if it was exposed in the future, the enumeration items could be confusing.

Anyway, I'm not against summarizing the options in an enumeration considering the current available configurations. So can we proceed with the change to the enumerated property?

Reply:
@yrodiere, following the suggestion to use an enumeration, I propose some adjustments to the names of the options:

```properties
quarkus.hibernate-orm.quote-identifiers = none
quarkus.hibernate-orm.quote-identifiers = all
quarkus.hibernate-orm.quote-identifiers = except-column-definitions
quarkus.hibernate-orm.quote-identifiers = only-keywords
```

I'll push with the new changes soon.

Reply:
@brunolmfg I suppose, if we name things right, we could be able to make that property multi-valued **in the future** (not now). E.g.:

```
quarkus.hibernate-orm.quote-identifiers = none
quarkus.hibernate-orm.quote-identifiers = all
quarkus.hibernate-orm.quote-identifiers = all-except-column-definitions
quarkus.hibernate-orm.quote-identifiers = keywords
quarkus.hibernate-orm.quote-identifiers = initial-underscore
quarkus.hibernate-orm.quote-identifiers = keywords,initial-underscore
```

We just need to avoid things like "only" or an unqualified "except" like in your suggestion, because those imply we're "disabling" some options, and that wouldn't be quite right with this hypothetical multi-valued property.

Though of course, that brings back the problem of some combinations not making sense (e.g. `all,none` or `all,keywords`). But it seems pretty clear to me why those don't make sense, and I think that would be obvious to inexperienced users as well, so maybe that's fine?

---

Alternatively, we could go with a structure that opens the way for more flags:

```
quarkus.hibernate-orm.quote-identifiers.strategy = none
quarkus.hibernate-orm.quote-identifiers.strategy = all
quarkus.hibernate-orm.quote-identifiers.strategy = all-except-column-definitions
quarkus.hibernate-orm.quote-identifiers.strategy = only-keywords
# To be added in the future if we really need to; not pretty, but it would work.
quarkus.hibernate-orm.quote-identifiers.quote-initial-underscore = true
```

---

WDYT?

Reply:
I prefer the latter structure because the multi-value alternative will return to the same problem with multiple boolean properties, i.e. useless combinations.

I'll change to the last structure naming, ok?